lib/dao.js

```diff
@@ -1565,20 +1565,23 @@ function coerceArray(val) {
  * @returns {Object} The coerced where clause
  * @private
  */
-DataAccessObject._coerce = function(where) {
+DataAccessObject._coerce = function(where, props) {
   var self = this;
   if (!where) {
     return where;
   }
 
+  if (!props) {
```

Comment:
Instead of this condition. Should do `props = props || self.definition.properties;`